pkg/azure/actuator.go

```diff
@@ -230,6 +236,52 @@ func (a *Actuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest) er
 		logger.Debug("credentials already up to date")
 		return nil
 	}
+	stsDetected, err := utils.IsTimedTokenCluster(a.client, ctx, logger)
+	if err != nil {
+		return err
+	}
+	if stsDetected {
+		logger.Debug("actuator detected Azure AD Workload Identity enabled cluster, enabling Workload Identity secret brokering for CredentialsRequests providing a Managed Identity")
+		azureProviderSpec, err := decodeProviderSpec(minterv1.Codec, cr)
+		if err != nil {
+			return err
+		}
+		azureFederatedTokenFile := cr.Spec.CloudTokenPath
+		if cr.Spec.CloudTokenPath == "" {
+			logger.Debugf("CredentialsRequest has no cloudTokenPath, defaulting azure_federated_token_file to %s", provisioning.OidcTokenPath)
+			azureFederatedTokenFile = provisioning.OidcTokenPath
+		}
+		// Check for old Manual Mode where all 4 fields are empty - defaulting to old behavior
+		// where CCO exists and the secret is created manually
+		if azureProviderSpec.AzureClientID == "" && azureProviderSpec.AzureTenantID == "" && azureProviderSpec.AzureSubscriptionID == "" && azureProviderSpec.AzureRegion == "" {
+			return nil
+		}
+		err = validateAzureProviderSpec(*azureProviderSpec)
```

Comment:
Note to reviewers: if some but not all fields required for this new manual mode workload identity workflow are set then we should error